src/test/Makefile

```diff
@@ -481,17 +481,12 @@ GTEST_DIR = ../../lib/test/gtest
 
 # Use clang/clang++ by default
 
-CC  := clang-12
-CXX := clang++-12
-ifneq ($(OSFAMILY), linux)
-ifeq ($(shell which $(CC) 2>/dev/null),)
-$(info Falling back to 'clang' on Windows and OSX.)
-CC  := clang
-CXX := clang++
+ifeq ($(origin CC), default)
```

Comment:
Sorry, this in not necessary (unless we want to override by environment variables).
`make test CC=clang-15 CXX=clang++-15` will take precedence over assignment, so `CC := clang-12` is fine